doc(dhcp): add features and limitations sections (#701)
harvester/docs: docs/advanced/addons/managed-dhcp.md

Harvester uses the planned infrastructure network so you must ensure that network connectivity is available and plan the IP pools in advance.
:::

## Unique Features

> - DHCP leases are stored in etcd as the single source of truth across the entire cluster
> - Each of the leases is static by nature and plays nicely with your current network infrastructure
> - The managed DHCP agents can still serve DHCP requests for existing entities even if the cluster's control plane stops working, ensuring your VM workload's network availability

Reviewer: ```suggestion
- DHCP leases are stored in etcd as the single source of truth across the entire cluster.
- Each of the leases is static by nature and works well with your current network infrastructure.
- The Managed DHCP agents can still serve DHCP requests for existing entities even if the cluster's control plane stops working, ensuring that your virtual machine workload's network remains available.
```

## Limitations

> - The managed DHCP does not support any network interfaces created inside the virtual machine. It only works with the network interfaces specified in the VirtualMachine CRs.
> - Adding or removing any network interfaces after the virtual machine is created will not get IP addresses allocated/deallocated because the actual MAC addresses are recorded in the VirtualMachineNetworkConfig CRs.
> - The DHCP RELEASE operation is currently not supported.
> - Updating IPPool configurations requires manually restarting the relevant agent Pods to take effect.

Reviewer: ```suggestion
- The Managed DHCP feature only works with the network interfaces specified in the VirtualMachine CRs. Network interfaces created in the virtual machine are not supported.
- IP addresses are not allocated or deallocated when you add or remove network interfaces after the virtual machine is created. The actual MAC addresses are recorded in the VirtualMachineNetworkConfig CRs.
- The DHCP RELEASE operation is currently not supported.
- IPPool configuration updates take effect only after you manually restart the relevant agent pods.
```
